src/config/into_document.rs

```diff
@@ -182,11 +182,19 @@ fn config_document(config: &Config) -> ServiceDocument {
             extend: false,
             description: None,
             name: pos(Name::new(type_name.clone())),
-            directives: type_def
-                .added_fields
-                .iter()
-                .map(|added_field| pos(added_field.to_directive()))
-                .collect::<Vec<_>>(),
+            directives: [
+                type_def
+                    .added_fields
+                    .iter()
+                    .map(|added_field: &super::AddField| pos(added_field.to_directive()))
+                    .collect::<Vec<_>>(),
+                if let Some(cache) = &type_def.cache {
+                    vec![pos(cache.to_directive())]
+                } else {
+                    vec![]
+                },
+            ]
+            .concat(),
```

Comment:
Could be

```suggestion
            directives: type_def
                .added_fields
                .iter()
                .map(|added_field: &super::AddField| pos(added_field.to_directive()))
                .chain(
                    type_def
                        .cache
                        .as_ref()
                        .map(|cache| pos(cache.to_directive())),
                )
                .collect::<Vec<_>>(),
```

But I wonder if that changes are really required. I don't think we have a way to specify cache for addedField with current SDL and there is no test case for this?

Reply:
It came up in #1054, which does test for identity for http_spec too, where there's a test for this.

Reply:
Also this is not the cache specified for the addedField. This is the cache specified for the type.

Reply:
> Also this is not the cache specified for the addedField. This is the cache specified for the type.

yes, you're right, it's the type itself. Can we extend the test for cache on type as well?

Reply:
There is no `graphql_spec` test for either (type or field) currently. `execution_spec` will check both.

Reply:
My bad. Brain isn't working great today :D On it.

Reply:
Huh. Caching was added in an... odd way, so specifying a `@cache` on the type doesn't apply it to the type itself, but to the fields directly. This means that the identity check fails:

```diff
  schema @server @upstream(baseURL: "http://jsonplacheholder.typicode.com") {
    query: Query
  }
 
  type Query {
    user: User @http(path: "/foo") @cache(maxAge: 300)
  }
 
< type User {
<   id: Int @cache(maxAge: 900)
<   name: String @cache(maxAge: 900)
> type User @cache(maxAge: 900) {
>   name: String
>   id: Int
  }
 ```

Should I re-write `@cache` logic to apply it to the type? Or should I write logic in `into_document` to assume type caching if every field has the same directive?